addons/l10n_id_pos/static/src/js/qr_code_popup.js

```diff
@@ -0,0 +1,40 @@
+import {QRPopup} from "@point_of_sale/app/utils/qr_code_popup/qr_code_popup";
+import {patch} from "@web/core/utils/patch";
+import {AlertDialog} from "@web/core/confirmation_dialog/confirmation_dialog";
+import {_t} from "@web/core/l10n/translation";
+
+patch(QRPopup.prototype, {
+    async _confirm() {
+        // Verify whether the payment has been recieved by QRIS
+
+        this.setButtonsDisabled(true);
+
+        const pm_line = this.props.line;
+        let result;
+
+        try {
+            result = await this.props.data.call("pos.payment.method", "l10n_id_verify_qris_status", [
+                [pm_line.payment_method_id.id],
+                pm_line.pos_order_id.uuid,
+            ]);
+        } catch (error) {
+            this.env.services.dialog.add(AlertDialog, {
+                title: _t("Failure"),
+                body: _t("Failure to verify QRIS payment status"),
+            });
+            this.setButtonsDisabled(false);
+            return false;
+        }
+
+        if (!result) {
+            this.env.services.dialog.add(AlertDialog, {
+                title: _t("Payment Status Update"),
+                body: _t("Payment Status returns unpaid"),
+            });
+            this.setButtonsDisabled(false);
+            return;
```

Comment:
`_confirm` method needs to return boolean -- determines whether the popup should close or not. Return `true` here if the QRPopup should close at this point, otherwise, return `false`.